tools/cli/workflow.go

```diff
@@ -364,3 +371,105 @@ func newActivityCommands() []cli.Command {
 		},
 	}
 }
+
+func newBatchCommands() []cli.Command {
+	return []cli.Command{
+		{
+			Name:    "describe",
+			Aliases: []string{"desc"},
+			Usage:   "Describe a batch operation job",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  FlagWorkflowIDWithAlias,
+					Usage: "WorkflowID",
+				},
+			},
+			Action: func(c *cli.Context) {
+				DescribeBatchJob(c)
+			},
+		},
+		{
+			Name:    "list",
+			Aliases: []string{"l"},
+			Usage:   "Describe a batch operation job",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  FlagOperator,
+					Usage: "Operator name that started the batch jobs",
+				},
+				cli.IntFlag{
+					Name:  FlagPageSizeWithAlias,
+					Value: 30,
+					Usage: "Result page size",
+				},
+			},
+			Action: func(c *cli.Context) {
+				ListBatchJobs(c)
+			},
+		},
+		{
+			Name:  "start",
+			Usage: "Start a batch operation job",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  FlagListQuery,
+					Usage: "Query to get workflows for being executed this batch operation",
+				},
+				cli.StringFlag{
+					Name:  FlagReasonWithAlias,
+					Usage: "Reason to run this batch job",
+				},
+				cli.StringFlag{
+					Name:  FlagBatchTypeWithAlias,
+					Usage: "Types supported: " + strings.Join(batcher.AllBatchTypes, ","),
+				},
+				//below are optional
+				cli.StringFlag{
+					Name:  FlagApplyOnChildren,
+					Value: "true",
+					Usage: "Whether apply onto children recursively. Only works with Terminate/Cancel",
+				},
+				cli.StringFlag{
+					Name:  FlagSignalNameWithAlias,
+					Usage: "Required for batch signal",
+				},
+				cli.StringFlag{
+					Name:  FlagInputWithAlias,
+					Usage: "Optional input of signal",
+				},
+				cli.IntFlag{
+					Name:  FlagRPS,
+					Value: batcher.DefaultRPS,
+					Usage: "RPS of processing",
+				},
+				cli.IntFlag{
+					Name:  FlagConcurrency,
+					Value: batcher.DefaultConcurrency,
+					Usage: "Number of goroutines of concurrent processing",
+				},
+				cli.IntFlag{
+					Name:  FlagAttemptsOnRetryableError,
+					Value: batcher.DefaultAttemptsOnRetryableError,
+					Usage: "Number of attempts on retryable error before giving up",
+				},
+				cli.StringSliceFlag{
+					Name:  FlagNonRetryableError,
+					Usage: "Errors that are not retryable",
+				},
+				cli.DurationFlag{
+					Name:  FlagHeartbeatTimeout,
+					Value: batcher.DefaultActivityHeartBeatTimeout,
+					Usage: "heartbeat timeout for the system workflow activity",
+				},
+				cli.DurationFlag{
+					Name:  FlagExecutionTimeout,
+					Value: 6 * time.Hour,
+					Usage: "how long the batch job should run",
```

Comment:
Should be hidden

Reply:
Deleted. Batch job will run forever if it needs to.